### ISSUE TITLE
Clarification on Baseline Settings

### ISSUE THREAD

**Issue by SII-ZhangYiFei** [NVIDIA/Isaac-GR00T#432]
Hi authors,
Thank you for releasing the code and paper — I’m currently trying to reproduce the baseline results reported in the paper.

However, even after following the repository instructions and the hyperparameters described in the paper, my reproduced results are significantly lower than the reported baseline numbers. Before I dig deeper, I would like to confirm whether I’m using the correct experimental settings.

🔍 My Reproduction Setup
Dataset

Using the dataset from: https://huggingface.co/GR00T-Dateset

Selected 100 demonstrations per task

Trained one model per task (not multi-task training)

Codebase

Using robomimic and robocasa

Baselines attempted:

BC Transformer w/ GMM (default robomimic configs)

```json
{
    "algo_name": "bc",
    "experiment": {
        "name": "seed_123_ds_CloseDoubleDoor",
        "validate": false,
        "logging": {
            "terminal_output_to_txt": true,
            "log_tb": true,
            "log_wandb": false,
            "wandb_proj_name": "bc_xfmr_10-20-None_robocasa_im"
        },
        "mse": {
            "enabled": false,
            "every_n_epochs": 50,
            "on_save_ckpt": true,
            "num_samples": 20,
            "visualize": true
        },
        "save": {
            "enabled": true,
            "every_n_seconds": null,
            "every_n_epochs": 100,
            "epochs": [],
            "on_best_validation": false,
            "on_best_rollout_return": false,
            "on_best_rollout_success_rate": false
        },
        "epoch_every_n_steps": 671,
        "validation_epoch_every_n_steps": 10,
        "env": null,
        "additional_envs": null,
        "render": false,
        "render_video": true,
        "keep_all_videos": true,
        "video_skip": 5,
        "rollout": {
            "enabled": true,
            "n": 100,
            "horizon": 1000,
            "rate": 100,
            "warmstart": 100,
            "terminate_on_success": true,
            "batched": true,
            "num_batch_envs": 20
        },
        "env_meta_update_dict": {
            "env_kwargs": {
                "generative_textures": null,
                "scene_split": null,
                "style_ids": null,
                "layout_ids": null,
                "layout_and_style_ids": [
                    [
                        1,
                        1
                    ],
                    [
                        2,
                        2
                    ],
                    [
                        4,
                        4
                    ],
                    [
                        6,
                        9
                    ],
                    [
                        7,
                        10
                    ]
                ],
                "randomize_cameras": false,
                "obj_instance_split": "B"
            }
        },
        "ckpt_path": null
    },
    "train": {
        "data": [
            {
                "path": "robomimic/datasets/CloseDoubleDoor.hdf5",
                "horizon": 430,
                "do_eval": true,
                "filter_key": null,
                "lang": "close the cabinet doors"
            }
        ],
        "output_dir": "robomimic/expdata/robocasa_76/bc_xfmr_gmm/CloseDoubleDoor",
        "num_data_workers": 32,
        "hdf5_cache_mode": "low_dim",
        "hdf5_use_swmr": true,
        "hdf5_load_next_obs": false,
        "hdf5_normalize_obs": false,
        "hdf5_filter_key": null,
        "hdf5_validation_filter_key": null,
        "seq_length": 10,
        "pad_seq_length": true,
        "frame_stack": 10,
        "pad_frame_stack": true,
        "dataset_keys": [
            "action_dict/rel_pos",
            "action_dict/rel_rot_axis_angle",
            "action_dict/gripper",
            "action_dict/base_mode"
        ],
        "action_keys": [
            "action_dict/rel_pos",
            "action_dict/rel_rot_axis_angle",
            "action_dict/gripper",
            "action_dict/base_mode"
        ],
        "action_config": {
            "actions": {
                "normalization": null
            },
            "actions_abs": {
                "normalization": "min_max"
            },
            "action_dict/abs_pos": {
                "normalization": "min_max"
            },
            "action_dict/abs_rot_axis_angle": {
                "normalization": "min_max",
                "format": "rot_axis_angle"
            },
            "action_dict/abs_rot_6d": {
                "normalization": null,
                "format": "rot_6d"
            },
            "action_dict/rel_pos": {
                "normalization": null
            },
            "action_dict/rel_rot_axis_angle": {
                "normalization": null,
                "format": "rot_axis_angle"
            },
            "action_dict/rel_rot_6d": {
                "normalization": null,
                "format": "rot_6d"
            },
            "action_dict/gripper": {
                "normalization": null
            },
            "action_dict/base_mode": {
                "normalization": null
            }
        },
        "goal_mode": null,
        "cuda": true,
        "batch_size": 64,
        "num_epochs": 2000,
        "seed": 123,
        "max_grad_norm": 100.0,
        "data_format": "robomimic",
        "shuffled_obs_key_groups": null
    },
    "algo": {
        "optim_params": {
            "policy": {
                "optimizer_type": "adamw",
                "learning_rate": {
                    "initial": 0.0001,
                    "decay_factor": 1.0,
                    "epoch_schedule": [
                        100
                    ],
                    "scheduler_type": "constant_with_warmup"
                },
                "regularization": {
                    "L2": 0.01
                }
            }
        },
        "loss": {
            "l2_weight": 1.0,
            "l1_weight": 0.0,
            "cos_weight": 0.0
        },
        "actor_layer_dims": [],
        "gaussian": {
            "enabled": false,
            "fixed_std": false,
            "init_std": 0.1,
            "min_std": 0.01,
            "std_activation": "softplus",
            "low_noise_eval": true
        },
        "gmm": {
            "enabled": true,
            "num_modes": 5,
            "min_std": 0.005,
            "std_activation": "softplus",
            "low_noise_eval": true
        },
        "vae": {
            "enabled": false,
            "latent_dim": 14,
            "latent_clip": null,
            "kl_weight": 1.0,
            "decoder": {
                "is_conditioned": true,
                "reconstruction_sum_across_elements": false
            },
            "prior": {
                "learn": false,
                "is_conditioned": false,
                "use_gmm": false,
                "gmm_num_modes": 10,
                "gmm_learn_weights": false,
                "use_categorical": false,
                "categorical_dim": 10,
                "categorical_gumbel_softmax_hard": false,
                "categorical_init_temp": 1.0,
                "categorical_temp_anneal_step": 0.001,
                "categorical_min_temp": 0.3
            },
            "encoder_layer_dims": [
                300,
                400
            ],
            "decoder_layer_dims": [
                300,
                400
            ],
            "prior_layer_dims": [
                300,
                400
            ]
        },
        "rnn": {
            "enabled": false,
            "horizon": 10,
            "hidden_dim": 400,
            "rnn_type": "LSTM",
            "num_layers": 2,
            "open_loop": false,
            "kwargs": {
                "bidirectional": false
            }
        },
        "transformer": {
            "enabled": true,
            "context_length": 10,
            "embed_dim": 512,
            "num_layers": 6,
            "num_heads": 8,
            "emb_dropout": 0.1,
            "attn_dropout": 0.1,
            "block_output_dropout": 0.1,
            "sinusoidal_embedding": false,
            "activation": "gelu",
            "supervise_all_steps": true,
            "nn_parameter_for_timesteps": true,
            "pred_future_acs": true,
            "causal": false
        },
        "language_conditioned": false
    },
    "observation": {
        "modalities": {
            "obs": {
                "low_dim": [
                    "robot0_base_to_eef_pos",
                    "robot0_base_to_eef_quat",
                    "robot0_base_pos",
                    "robot0_base_quat",
                    "robot0_gripper_qpos"
                ],
                "rgb": [
                    "robot0_agentview_left_image",
                    "robot0_agentview_right_image",
                    "robot0_eye_in_hand_image"
                ],
                "depth": [],
                "scan": []
            },
            "goal": {
                "low_dim": [],
                "rgb": [],
                "depth": [],
                "scan": []
            }
        },
        "encoder": {
            "low_dim": {
                "core_class": null,
                "core_kwargs": {},
                "obs_randomizer_class": null,
                "obs_randomizer_kwargs": {}
            },
            "rgb": {
                "core_class": "VisualCoreLanguageConditioned",
                "core_kwargs": {
                    "feature_dimension": 64,
                    "backbone_class": "ResNet18ConvFiLM",
                    "backbone_kwargs": {
                        "pretrained": false,
                        "input_coord_conv": false
                    },
                    "pool_class": "SpatialSoftmax",
                    "pool_kwargs": {
                        "num_kp": 32,
                        "learnable_temperature": false,
                        "temperature": 1.0,
                        "noise_std": 0.0
                    }
                },
                "obs_randomizer_class": "CropRandomizer",
                "obs_randomizer_kwargs": {
                    "crop_height": 76,
                    "crop_width": 76,
                    "num_crops": 1,
                    "pos_enc": false
                }
            },
            "depth": {
                "core_class": "VisualCore",
                "core_kwargs": {},
                "obs_randomizer_class": null,
                "obs_randomizer_kwargs": {}
            },
            "scan": {
                "core_class": "ScanCore",
                "core_kwargs": {},
                "obs_randomizer_class": null,
                "obs_randomizer_kwargs": {}
            }
        }
    },
    "meta": {
        "hp_base_config_file": "robomimic/robomimic/exps/templates/bc_transformer.json",
        "hp_keys": [
            "seed",
            "ds",
            "obsrandargs"
        ],
        "hp_values": [
            123,
            "CloseDoubleDoor",
            {
                "crop_height": 76,
                "crop_width": 76,
                "num_crops": 1,
                "pos_enc": false
            }
        ]
    }
}
```

Diffusion Policy (default configs)

```json
{
    "algo_name": "diffusion_policy",
    "experiment": {
        "name": "seed_123_ds_CloseDoubleDoor",
        "validate": false,
        "logging": {
            "terminal_output_to_txt": true,
            "log_tb": true,
            "log_wandb": false,
            "wandb_proj_name": "diffusion_policy_10-30-dp_1030_robocasa_im"
        },
        "mse": {
            "enabled": false,
            "every_n_epochs": 50,
            "on_save_ckpt": true,
            "num_samples": 20,
            "visualize": true
        },
        "save": {
            "enabled": true,
            "every_n_seconds": null,
            "every_n_epochs": 100,
            "epochs": [],
            "on_best_validation": false,
            "on_best_rollout_return": false,
            "on_best_rollout_success_rate": false
        },
        "epoch_every_n_steps": 671,
        "validation_epoch_every_n_steps": 10,
        "env": null,
        "additional_envs": null,
        "render": false,
        "render_video": true,
        "keep_all_videos": true,
        "video_skip": 5,
        "rollout": {
            "enabled": true,
            "n": 100,
            "horizon": 630,
            "rate": 100,
            "warmstart": 100,
            "terminate_on_success": true,
            "batched": true,
            "num_batch_envs": 20
        },
        "env_meta_update_dict": {
            "env_kwargs": {
                "generative_textures": null,
                "scene_split": null,
                "style_ids": null,
                "layout_ids": null,
                "layout_and_style_ids": [
                    [
                        1,
                        1
                    ],
                    [
                        2,
                        2
                    ],
                    [
                        4,
                        4
                    ],
                    [
                        6,
                        9
                    ],
                    [
                        7,
                        10
                    ]
                ],
                "randomize_cameras": false,
                "obj_instance_split": "B",
                "controller_configs": {
                    "control_delta": false
                }
            }
        },
        "ckpt_path": null
    },
    "train": {
        "data": [
            {
                "path": "robomimic/datasets/CloseDoubleDoor.hdf5",
                "horizon": 430,
                "do_eval": true,
                "filter_key": null,
                "lang": "close the cabinet doors"
            }
        ],
        "output_dir": "robomimic/expdata/robocasa_76/dp/CloseDoubleDoor",
        "num_data_workers": 32,
        "hdf5_cache_mode": "low_dim",
        "hdf5_use_swmr": true,
        "hdf5_load_next_obs": false,
        "hdf5_normalize_obs": false,
        "hdf5_filter_key": null,
        "hdf5_validation_filter_key": null,
        "seq_length": 15,
        "pad_seq_length": true,
        "frame_stack": 2,
        "pad_frame_stack": true,
        "dataset_keys": [
            "action_dict/rel_pos",
            "action_dict/rel_rot_axis_angle",
            "action_dict/gripper",
            "action_dict/base_mode"
        ],
        "action_keys": [
            "action_dict/rel_pos",
            "action_dict/rel_rot_axis_angle",
            "action_dict/gripper",
            "action_dict/base_mode"
        ],
        "action_config": {
            "actions": {
                "normalization": null
            },
            "actions_abs": {
                "normalization": "min_max"
            },
            "action_dict/abs_pos": {
                "normalization": "min_max"
            },
            "action_dict/abs_rot_axis_angle": {
                "normalization": "min_max",
                "format": "rot_axis_angle"
            },
            "action_dict/abs_rot_6d": {
                "normalization": null,
                "format": "rot_6d"
            },
            "action_dict/rel_pos": {
                "normalization": null
            },
            "action_dict/rel_rot_axis_angle": {
                "normalization": null,
                "format": "rot_axis_angle"
            },
            "action_dict/rel_rot_6d": {
                "normalization": null,
                "format": "rot_6d"
            },
            "action_dict/gripper": {
                "normalization": null
            },
            "action_dict/base_mode": {
                "normalization": null
            }
        },
        "goal_mode": null,
        "cuda": true,
        "batch_size": 64,
        "num_epochs": 2000,
        "seed": 123,
        "max_grad_norm": null,
        "data_format": "robomimic",
        "shuffled_obs_key_groups": null
    },
    "algo": {
        "optim_params": {
            "policy": {
                "learning_rate": {
                    "initial": 0.0001,
                    "decay_factor": 0.1,
                    "epoch_schedule": [
                        100
                    ],
                    "scheduler_type": "constant"
                },
                "regularization": {
                    "L2": 0.0
                }
            }
        },
        "horizon": {
            "observation_horizon": 2,
            "action_horizon": 8,
            "prediction_horizon": 16
        },
        "unet": {
            "enabled": true,
            "diffusion_step_embed_dim": 256,
            "down_dims": [
                256,
                512,
                1024
            ],
            "kernel_size": 5,
            "n_groups": 8
        },
        "ema": {
            "enabled": true,
            "power": 0.75
        },
        "language_conditioned": false,
        "ddpm": {
            "enabled": false,
            "num_train_timesteps": 100,
            "num_inference_timesteps": 100,
            "beta_schedule": "squaredcos_cap_v2",
            "clip_sample": true,
            "prediction_type": "epsilon"
        },
        "ddim": {
            "enabled": true,
            "num_train_timesteps": 100,
            "num_inference_timesteps": 10,
            "beta_schedule": "squaredcos_cap_v2",
            "clip_sample": true,
            "set_alpha_to_one": true,
            "steps_offset": 0,
            "prediction_type": "epsilon"
        }
    },
    "observation": {
        "modalities": {
            "obs": {
                "low_dim": [
                    "robot0_base_to_eef_pos",
                    "robot0_base_to_eef_quat",
                    "robot0_base_pos",
                    "robot0_base_quat",
                    "robot0_gripper_qpos"
                ],
                "rgb": [
                    "robot0_agentview_left_image",
                    "robot0_agentview_right_image",
                    "robot0_eye_in_hand_image"
                ],
                "depth": [],
                "scan": []
            },
            "goal": {
                "low_dim": [],
                "rgb": [],
                "depth": [],
                "scan": []
            }
        },
        "encoder": {
            "low_dim": {
                "core_class": null,
                "core_kwargs": {},
                "obs_randomizer_class": null,
                "obs_randomizer_kwargs": {}
            },
            "rgb": {
                "core_class": "VisualCoreLanguageConditioned",
                "core_kwargs": {
                    "feature_dimension": 64,
                    "backbone_class": "ResNet18ConvFiLM",
                    "backbone_kwargs": {
                        "pretrained": false,
                        "input_coord_conv": false
                    },
                    "pool_class": "SpatialSoftmax",
                    "pool_kwargs": {
                        "num_kp": 32,
                        "learnable_temperature": false,
                        "temperature": 1.0,
                        "noise_std": 0.0
                    }
                },
                "obs_randomizer_class": "CropRandomizer",
                "obs_randomizer_kwargs": {
                    "crop_height": 76,
                    "crop_width": 76,
                    "num_crops": 1,
                    "pos_enc": false
                }
            },
            "depth": {
                "core_class": "VisualCore",
                "core_kwargs": {},
                "obs_randomizer_class": null,
                "obs_randomizer_kwargs": {}
            },
            "scan": {
                "core_class": "ScanCore",
                "core_kwargs": {},
                "obs_randomizer_class": null,
                "obs_randomizer_kwargs": {}
            }
        }
    },
    "meta": {
        "hp_base_config_file": "robomimic/robomimic/exps/templates/diffusion_policy.json",
        "hp_keys": [
            "seed",
            "ds",
            "ac_keys",
            "obsrandargs"
        ],
        "hp_values": [
            123,
            "CloseDoubleDoor",
            "abs_acs",
            {
                "crop_height": 76,
                "crop_width": 76,
                "num_crops": 1,
                "pos_enc": false
            }
        ]
    }
}
```

Training

Used default robomimic hyperparameters

Trained each task for 1000 epochs

Evaluation

Rolled out 100 episodes per task

Evaluation environments: GR1 and RoboCasa tasks

I got 21% SR for BC Transformer w/ GMM on robocasa tasks and totally zero on gr1 table tasks
16% SR for Diffusion Policy on robocasa tasks.

❓ Questions

To better understand the gap, could you please clarify the following?

Did you train one model per task, or a single multi-task model across all tasks?
(This alone could significantly change results.)

Could you share the exact config files used for the reported baselines?

Model architecture params

GMM settings

Transformer settings

Diffusion Policy network details

Optimizer, learning rate schedules, batch size, etc.

Evaluation details

What was the exact rollout setup?

Were there specific seeds or environment settings?

If possible, could you provide:

A reference training config

Or a training checkpoint from the baseline models

🙏 Thank you

I appreciate your work and would be grateful for any additional information that could help reproduce the baseline performance accurately. Please let me know if there is any other detail I should provide to help debug.

**Comment by SII-ZhangYiFei** [comment]
Hi, just a friendly follow-up.
Is there any update on this issue, or anything I can do to help reproduce it?

Thank you!

**Comment by squarefk** [comment]
@Dingry Could you please share some advice on training BC-Transformers? One thing we discussed is that we may need to convert from joint-space actions to end-effector actions to make BC-Transformers work. We did observe poor performance with the prior.

**Comment by artpli** [comment]
Hi @Dingry, could you please share any experiences here?
